drivers/gpio/gpio_mspm0.c

```diff
@@ -0,0 +1,312 @@
+
+	return 0;
+}
+
+static int gpio_mspm0_port_set_masked_raw(const struct device *port,
+					  uint32_t mask, uint32_t value)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_writePinsVal(config->base, mask, value);
+
+	return 0;
+}
+
+static int gpio_mspm0_port_set_bits_raw(const struct device *port,
+					uint32_t mask)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_setPins(config->base, mask);
+
+	return 0;
+}
+
+static int gpio_mspm0_port_clear_bits_raw(const struct device *port,
+					  uint32_t mask)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_clearPins(config->base, mask);
+
+	return 0;
+}
+
+static int gpio_mspm0_port_toggle_bits(const struct device *port,
+				       uint32_t mask)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_togglePins(config->base, mask);
+
+	return 0;
+}
+
+static int gpio_mspm0_pin_configure(const struct device *port,
+				    gpio_pin_t pin,
+				    gpio_flags_t flags)
+{
+	const struct gpio_mspm0_config *config = port->config;
+	/* determine pull up resistor value based on flags */
+	DL_GPIO_RESISTOR pull_res;
+
+	if (flags & GPIO_PULL_UP) {
+		pull_res = DL_GPIO_RESISTOR_PULL_UP;
+	} else if (flags & GPIO_PULL_DOWN) {
+		pull_res = DL_GPIO_RESISTOR_PULL_DOWN;
+	} else {
+		pull_res = DL_GPIO_RESISTOR_NONE;
+	}
+
+	/* Config pin based on flags */
+	switch (flags & (GPIO_INPUT | GPIO_OUTPUT)) {
+	case GPIO_INPUT:
+		DL_GPIO_initDigitalInputFeatures(config->pincm_lut[pin],
+						 DL_GPIO_INVERSION_DISABLE,
+						 pull_res,
+						 DL_GPIO_HYSTERESIS_DISABLE,
+						 DL_GPIO_WAKEUP_DISABLE);
+		break;
+	case GPIO_OUTPUT:
+		DL_GPIO_initDigitalOutputFeatures(config->pincm_lut[pin],
+						  DL_GPIO_INVERSION_DISABLE,
+						  pull_res,
+						  DL_GPIO_DRIVE_STRENGTH_LOW,
+						  DL_GPIO_HIZ_DISABLE);
+
+		/* Set initial state */
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			gpio_mspm0_port_set_bits_raw(port, BIT(pin));
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			gpio_mspm0_port_clear_bits_raw(port, BIT(pin));
+		}
+		/* Enable output */
+		DL_GPIO_enableOutput(config->base, BIT(pin));
+		break;
+	case GPIO_DISCONNECTED:
+		DL_GPIO_disableOutput(config->base, BIT(pin));
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int gpio_mspm0_pin_interrupt_configure(const struct device *port,
+					      gpio_pin_t pin,
+					      enum gpio_int_mode mode,
+					      enum gpio_int_trig trig)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	/* Config interrupt */
+	switch (mode) {
+	case GPIO_INT_MODE_DISABLED:
+		DL_GPIO_clearInterruptStatus(config->base, BIT(pin));
+		DL_GPIO_disableInterrupt(config->base, BIT(pin));
+		break;
+	case GPIO_INT_MODE_EDGE:
+		uint32_t polarity = 0x00;
+
+		if (trig & GPIO_INT_TRIG_LOW) {
+			polarity |= BIT(0);
+		}
+
+		if (trig & GPIO_INT_TRIG_HIGH) {
+			polarity |= BIT(1);
+		}
+
+		if (pin < 16) {
```

Comment:
magic value 16. 
Make it into a macro and explain why it's handled differently.